src/vs/platform/telemetry/node/workbenchCommonProperties.ts

```diff
@@ -51,3 +53,20 @@ export function resolveWorkbenchCommonProperties(storageService: IStorageService
 // 	storageService.store('telemetry.instanceId', result);
 // 	return result;
 // }
+
+// {{SQL CARBON EDIT}}
+function setUsageDates(storageService: IStorageService): void {
+	// daily last usage date
+	const appStartDate = new Date('January 1, 2000');
+	const dailyLastUseDate = storageService.get('telemetry.dailyLastUseDate') || appStartDate;
+	storageService.store('telemetry.dailyLastUseDate', dailyLastUseDate);
+
+	// weekly last usage date
+	const weeklyLastUseDate = storageService.get('telemetry.weeklyLastUseDate') || new Date().toUTCString();
```

Comment:
why don't we initialize to `appStartDate` here as well?  Don't we want to send weekly and monthly events on 1st launch as week?